pkg/controller/elasticsearch/validation/validations.go

```diff
@@ -33,6 +33,20 @@ var Validations = []Validation{
 	noBlacklistedSettings,
 	validSanIP,
 	pvcModification,
+	checkNodeSetNameUniqueness,
+}
+
+func checkNodeSetNameUniqueness(ctx Context) validation.Result {
+	nodeSets := ctx.Proposed.Elasticsearch.Spec.NodeSets
+	names := make(map[string]interface{})
```

Comment:
It doesn't matter here much but `struct{}` is better than `interface{}` because it has a size of 0.